optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/common/BavetNodeBuildPolicy.java

```diff
@@ -77,4 +77,22 @@ public int getNodeOrderMaximum() {
         return joinConstraintStreamToJoinBridgeNodeMap;
     }
 
+    public List<BavetNode> getCreatedNodes() {
+        // Make a sequential list of unique nodes.
+        SortedMap<Integer, BavetNode> nodeIndexToNodeMap = Stream.concat(sharableNodeMap.keySet().stream(),
+                constraintIdToScoringNodeMap.values().stream())
+                .distinct()
+                .collect(Collectors.toMap(k -> k.getNodeIndex(), Function.identity(), (a, b) -> {
+                    throw new IllegalStateException("Impossible state: nodes (" + a + ") and (" + b + ") share index.");
```

Comment:
```suggestion
                    throw new IllegalStateException("Impossible state: 2 nodes (" + a + ", " + b + ") share the same index (" + a.getNodeIndex() + ").");
```

Reply:
Motivation: grammar ("the", etc) and show the node index in that error message if we have it available anyway.